examples/native-fungible/src/service.rs

```diff
@@ -35,28 +36,42 @@ impl Service for NativeFungibleToken {
         _context: &QueryContext,
         request: Request,
     ) -> Result<Response, Self::Error> {
-        let schema =
-            Schema::build(QueryRoot, Operation::mutation_root(), EmptySubscription).finish();
+        let schema = Schema::build(
+            QueryRoot {
+                context: self.context().clone(),
+            },
+            Operation::mutation_root(),
+            EmptySubscription,
+        )
+        .finish();
         let response = schema.execute(request).await;
         Ok(response)
     }
 }
 
-struct QueryRoot;
+struct QueryRoot {
+    context: ViewStorageContext,
+}
 
 #[Object]
 impl QueryRoot {
-    async fn balance(&self, owner: AccountOwner) -> Result<Amount, async_graphql::Error> {
-        let owner = match owner {
-            AccountOwner::User(owner) => owner,
-            AccountOwner::Application(_) => panic!("Applications not supported yet!"),
-        };
-        Ok(system_api::current_owner_balance(owner))
-    }
-
     async fn ticker_symbol(&self) -> Result<String, async_graphql::Error> {
         Ok(String::from(TICKER_SYMBOL))
     }
+
+    async fn accounts(&self) -> Result<MapView<AccountOwner, Amount>, async_graphql::Error> {
+        let mut accounts: MapView<AccountOwner, Amount> =
+            MapView::load(self.context.clone()).await?;
```

Comment:
(So this always creates an _emtpy_ `MapView` because our own contract state happens to be empty, right? I wonder whether it would still be safer to just add an actual `MapView` to the state, which is never written to. That way we're not using the same base key for both the contract state and a `MapView`.)

Reply:
But I do want an empty `MapView` there 🤔
Do you mean that I could create the `MapView` in the state, and always overwrite it here and return it?
Also, what do you mean by "base key"?

Reply:
I agree. But I think we don't need to add it to the state, since we're also considering writing a custom type with only the requested entries. I wonder if a `BTreeMap` or something already implements an equivalent GraphQL interface :thinking: I tried looking at the `async-graphql` documentation, but I don't understand enough of GraphQL to know which trait I'm looking for :sweat: